geodata/models/common.py

```diff
@@ -0,0 +1,73 @@
+import tempfile
+
+from django.conf import settings
+from django.contrib.gis.db import models
+from django.utils import timezone
+
+
+class DeferredFieldsManager(models.Manager):
+    def __init__(self, *deferred_fields):
+        self.deferred_fields = deferred_fields
+        super().__init__()
+
+    def get_queryset(self):
+        return super().get_queryset().defer(*self.deferred_fields)
+
+
+class ModifiableEntry(models.Model):
+    """A base class for models that need to track modified datetimes."""
+
+    modified = models.DateTimeField(editable=False, help_text='The last time this entry was saved.')
+    created = models.DateTimeField(editable=False, help_text='When this was added to the database.')
+
+    class Meta:
+        abstract = True
+
+    def save(self, *args, **kwargs):
+        if not self.id:
+            self.created = timezone.now()
+        self.modified = timezone.now()
+        super(ModifiableEntry, self).save(*args, **kwargs)
+
+
+class SpatialEntry(ModifiableEntry):
+    """Common model to all geospatial data entries."""
+
+    name = models.CharField(max_length=100, blank=True, null=True)
+    # An optional description field in case the user needs to add context
+    description = models.TextField(blank=True, null=True)
+    # Datetime of creation for the dataset
+    acquisition_date = models.DateTimeField(default=timezone.now)
```

Comment:
I think we want the acquisition_date to be nullable and default to null.  If we do a search based on acquisition time, it could be misleading to have a recent time here when the data could be much older.